_docs/render.py

```diff
@@ -0,0 +1,196 @@
+
+    needed = _get_needed_types(source)
+    lines = [v for k, v in type_strings().items() if k in needed]
+    if lines:
+        lines.extend(["", ""])
+    lines.extend(source.splitlines())
+    return "\n".join(lines)
+
+
+def example_implementation(contrib_name: str) -> str:
+    """Build an example string of python source implementing a specific contribution."""
+    contrib = getattr(EXAMPLE_MANIFEST.contributions, contrib_name)
+    if isinstance(contrib, list):
+        return "\n\n".join([_build_example(x) for x in contrib]).strip()
+    return _build_example(contrib)
+
+
+def example_contribution(
+    contrib_name: str, format="yaml", manifest: PluginManifest = EXAMPLE_MANIFEST
+) -> str:
+    """Get small manifest example for just contribution named `contrib_name`"""
+    assert manifest.contributions
+    contribs = getattr(manifest.contributions, contrib_name)
+    # only take the first command example ... the rest are for executables
+    if contrib_name == "commands":
+        contribs = [contribs[0]]
+
+    ex = ContributionPoints(**{contrib_name: contribs})
+    # for "executables", include associated command
+    ExampleCommands = manifest.contributions.commands
+    assert ExampleCommands
+    for c in contribs or ():
+        if isinstance(c, Executable):
+            associated_command = next(i for i in ExampleCommands if i.id == c.command)
+            if not ex.commands:
+                ex.commands = []
+            ex.commands.append(associated_command)
+    output = {"contributions": json.loads(ex.json(exclude_unset=True))}
+    if format == "yaml":
+        return yaml.safe_dump(output, sort_keys=False)
+    if format == "toml":
+        import pytomlpp as toml
+
+        return toml.dumps(output)
+    if format == "json":
+        return json.dumps(output)
+    raise ValueError("Invalid format: {format}.  Must be 'yaml', 'toml' or 'json'.")
+
+
+def main(dest: Path = _BUILD):
+    """Render all jinja docs in ./templates and output to `dest`"""
+
+    # register the example plugin so we can use `.get_callable()` in _build_example
+    sys.path.append(str(DOCS.absolute()))
+    PluginManager.instance().register(EXAMPLE_MANIFEST)
+
+    this = Path(__file__).parent.name
+    env = Environment(loader=PackageLoader(this), autoescape=select_autoescape())
+    env.filters["example_contribution"] = example_contribution
+    env.filters["example_implementation"] = example_implementation
+
+    dest.mkdir(exist_ok=True, parents=True)
+    schema = PluginManifest.schema()
+    contributions = schema["definitions"]["ContributionPoints"]["properties"]
+    context = {
+        "schema": schema,
+        "contributions": contributions,
+        "example": EXAMPLE_MANIFEST,
+        # "specs": _get_specs(),
+        "specs": {},
+    }
+
+    for t in TEMPLATES.glob("*.jinja"):
+        template = env.get_template(t.name)
+        _dest = dest / f"{t.stem}"
+        _dest.write_text(template.render(context))
```

Comment:
This should fix the windows failures

```suggestion
        _dest.write_text(template.render(context),encoding="utf-8")
```

Reply:
thanks!